### PR TITLE
Feature: get models

### PR DESCRIPTION
**Pull request by Zorin95670**
:information_source:  Don't forget to modify the changelog.md before merging this branch.

### CODE REVIEW

File: src/parser/DefaultParser.js [hunk 1]
```diff
@@ -29,6 +29,16 @@ class DefaultParser {
     this.pluginData.parseErrors = [];
   }
 
+  /**
+   * Get the list of models from all parsable files.
+   *
+   * @param {FileInformation[]} [files=[]] - List of parsable files.
+   * @return {FileInformation[]} List of Folder or file path that represents a model.
```
Comment (k-y0u) [inline src/parser/DefaultParser.js]: List of folder and file paths that represent a model.

File: src/parser/DefaultParser.js [hunk 2]
```diff
@@ -29,6 +29,16 @@ class DefaultParser {
     this.pluginData.parseErrors = [];
   }
 
+  /**
+   * Get the list of models from all parsable files.
```
Comment (k-y0u) [inline src/parser/DefaultParser.js:33]: Get the list of model paths from all parsable files.

File: tests/unit/parser/DefaultParser.spec.js [hunk 3]
```diff
@@ -19,42 +20,72 @@ describe('Test Class: DefaultParser()', () => {
...
+      expect(defaultParser.pluginData.components).toEqual([]);
+      expect(defaultParser.pluginData.parseErrors).toEqual([]);
     });
+  });
 
-    describe('Test method: isParsable', () => {
-      it('on any not null file name should return true', () => {
-        const defaultParser = new DefaultParser();
+  describe('Test method: isParsable', () => {
+    it('on any not null file name should return true', () => {
```
Comment (k-y0u) [inline tests/unit/parser/DefaultParser.spec.js:38]: rephrase

File: tests/unit/parser/DefaultParser.spec.js [hunk 4]
```diff
@@ -19,42 +20,72 @@ describe('Test Class: DefaultParser()', () => {
...
+      const defaultParser = new DefaultParser();
 
-        expect(defaultParser.isParsable('')).toBeTruthy();
-      });
+      expect(defaultParser.isParsable('')).toBeTruthy();
+    });
 
-      it('on null/undefined file name should return false', () => {
-        const defaultParser = new DefaultParser();
+    it('on null/undefined file name should return false', () => {
```
Comment (k-y0u) [inline tests/unit/parser/DefaultParser.spec.js:44]: rephrase

File: tests/unit/parser/DefaultParser.spec.js [hunk 5]
```diff
@@ -19,42 +20,72 @@ describe('Test Class: DefaultParser()', () => {
...
+
+      expect(defaultParser.getModels()).toEqual([]);
     });
 
-    it('Should save configuration even if the file content is null', () => {
+    it('should provided files array', () => {
+      const defaultParser = new DefaultParser();
+
+      expect(defaultParser.getModels([])).toEqual([]);
+      expect(defaultParser.getModels([new FileInformation({ path: 'a' })]))
```
Comment (k-y0u) [inline tests/unit/parser/DefaultParser.spec.js:63]: divide

File: tests/unit/parser/DefaultParser.spec.js [hunk 6]
```diff
@@ -19,42 +20,72 @@ describe('Test Class: DefaultParser()', () => {
...
+        .toEqual([new FileInformation({ path: 'a' })]);
+    });
+  });
+
+  it('Should save configuration even if the file content is null', () => {
```
Comment (k-y0u) [inline tests/unit/parser/DefaultParser.spec.js:68]: should

File: tests/unit/parser/DefaultParser.spec.js [hunk 7]
```diff
@@ -63,82 +94,66 @@ describe('Test Class: DefaultParser()', () => {
...
 
-    describe('Test method: parseConfiguration', () => {
-      it('Should not change component on empty configuration', () => {
-        const pluginData = new DefaultData({ name: 'test' });
-        const defaultParser = new DefaultParser(pluginData);
-        const configFile = new FileInput({
-          path: 'test.json',
-          content: JSON.stringify({}),
-        });
+    it('Should not change component without plugin configuration', () => {
```
Comment (k-y0u) [inline tests/unit/parser/DefaultParser.spec.js:97]: should

File: tests/unit/parser/DefaultParser.spec.js [hunk 8]
```diff
@@ -63,82 +94,66 @@ describe('Test Class: DefaultParser()', () => {
...
+      pluginData.components.push(new Component({ id: 'c2' }));
+      defaultParser.parseConfiguration(configFile);
 
-        pluginData.components.push(new Component({ id: 'c2' }));
-        defaultParser.parseConfiguration(configFile);
+      expect(pluginData.components).toEqual([new Component({ id: 'c2' })]);
+    });
 
-        expect(pluginData.components).toEqual([new Component({ id: 'c2' })]);
+    it('Should not change component without component configuration', () => {
```
Comment (k-y0u) [inline tests/unit/parser/DefaultParser.spec.js:115]: should

File: tests/unit/parser/DefaultParser.spec.js [hunk 9]
```diff
@@ -63,82 +94,66 @@ describe('Test Class: DefaultParser()', () => {
...
-
-        pluginData.components.push(new Component({ id: 'c1' }));
-        pluginData.components.push(new Component({ id: 'c2' }));
-        defaultParser.parseConfiguration(configFile);
-
-        expect(pluginData.components).toEqual([
-          new Component({ id: 'c1', drawOption: new ComponentDrawOption({ x: 1 }) }),
-          new Component({ id: 'c2', drawOption: new ComponentDrawOption({ x: 2 }) }),
-        ]);
+    it('Should update component', () => {
```
Comment (k-y0u) [inline tests/unit/parser/DefaultParser.spec.js:136]: should

Comment (k-y0u) [reply tests/unit/parser/DefaultParser.spec.js:136]: components

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/parser/DefaultParser.js (hunk 1 above)
Comment (Zorin95670) [reply src/parser/DefaultParser.js]: done

File: src/parser/DefaultParser.js (hunk 2 above)
Comment (Zorin95670) [reply src/parser/DefaultParser.js:33]: done

File: tests/unit/parser/DefaultParser.spec.js (hunk 5 above)
Comment (Zorin95670) [reply tests/unit/parser/DefaultParser.spec.js:63]: done

File: tests/unit/parser/DefaultParser.spec.js [hunk 10]
```diff
@@ -19,42 +20,72 @@ describe('Test Class: DefaultParser()', () => {
...
+      expect(defaultParser.getModels()).toEqual([]);
     });
 
-    it('Should save configuration even if the file content is null', () => {
+    it('should provided files array', () => {
+      const defaultParser = new DefaultParser();
+
+      expect(defaultParser.getModels([])).toEqual([]);
+      expect(defaultParser.getModels([new FileInformation({ path: 'a' })]))
+        .toEqual([new FileInformation({ path: 'a' })]);
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply tests/unit/parser/DefaultParser.spec.js]: done

File: tests/unit/parser/DefaultParser.spec.js [hunk 11]
```diff
@@ -19,42 +20,72 @@ describe('Test Class: DefaultParser()', () => {
...
+    });
+
+    pluginData.components.push(new Component({ id: 'c2' }));
+    defaultParser.parseConfiguration(configFile);
+
+    expect(pluginData.components).toEqual([new Component({ id: 'c2' })]);
+  });
+
+  describe('Test method: parseConfiguration', () => {
+    it('Should not change component on empty configuration', () => {
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply tests/unit/parser/DefaultParser.spec.js:83]: done

File: tests/unit/parser/DefaultParser.spec.js (hunk 6 above)
Comment (Zorin95670) [reply tests/unit/parser/DefaultParser.spec.js:68]: done

File: tests/unit/parser/DefaultParser.spec.js (hunk 7 above)
Comment (Zorin95670) [reply tests/unit/parser/DefaultParser.spec.js:97]: done

File: tests/unit/parser/DefaultParser.spec.js (hunk 8 above)
Comment (Zorin95670) [reply tests/unit/parser/DefaultParser.spec.js:115]: done

File: tests/unit/parser/DefaultParser.spec.js (hunk 9 above)
Comment (Zorin95670) [reply tests/unit/parser/DefaultParser.spec.js:136]: done

File: tests/unit/models/DefaultPlugin.spec.js [hunk 12]
```diff
@@ -49,184 +50,210 @@ describe('Test class: DefaultPlugin', () => {
...
-          pluginData: {
-            initLinkDefinitions: mockInitLinkDefinitions,
-            emitEvent: jest.fn(),
-          },
-          pluginMetadata: {
-            parse: mockParse,
-          },
-        });
+  describe('Test method: init', () => {
+    it('should set events and call validate and parse of metadata', () => {
```
Comment (Hundraw) [inline tests/unit/models/DefaultPlugin.spec.js:54]: I find the wording strange but I don't know what to suggest better.

File: tests/unit/parser/DefaultParser.spec.js (hunk 4 above)
Comment (Zorin95670) [reply tests/unit/parser/DefaultParser.spec.js]: done

File: tests/unit/models/DefaultPlugin.spec.js [hunk 13]
```diff
@@ -49,184 +50,210 @@ describe('Test class: DefaultPlugin', () => {
...
-        });
+      plugin.isParsable();
+      expect(mockIsParsable).toBeCalled();
+    });
+  });
 
-        plugin.isParsable();
-        expect(mockIsParsable).toBeCalled();
+  describe('Test method: getModels', () => {
+    it('should return empty array without files', () => {
```
Comment (Hundraw) [inline tests/unit/models/DefaultPlugin.spec.js:121]: should return an empty array if there are no files

**Review by k-y0u — APPROVED** [review]

File: tests/unit/models/DefaultPlugin.spec.js [hunk 14]
```diff
@@ -49,184 +50,210 @@ describe('Test class: DefaultPlugin', () => {
...
 
-    describe('Test method: draw', () => {
-      it('should call draw method from drawer', () => {
-        const mockDraw = jest.fn();
-        const plugin = new DefaultPlugin({
-          pluginDrawer: {
-            draw: mockDraw,
-          },
-        });
+    it('should return all wanted models', () => {
```
Comment (Hundraw) [inline tests/unit/models/DefaultPlugin.spec.js:133]: should return all parsable models

**Review by Hundraw — CHANGES_REQUESTED** [review]

**Review by k-y0u — APPROVED** [review]

**Pull request merged**